src/cache_padded.rs

```diff
@@ -1,120 +1,84 @@
-use std::marker;
-use std::cell::UnsafeCell;
 use std::fmt;
+use std::marker::PhantomData;
 use std::mem;
-use std::ptr;
 use std::ops::{Deref, DerefMut};
+use std::ptr;
 
-// For now, treat this as an arch-independent constant.
-const CACHE_LINE: usize = 32;
+/// 64 bytes is the most common cache line size on modern machines.
+const CACHE_LINE_BYTES: usize = 64;
 
-#[cfg_attr(feature = "nightly", repr(simd))]
-#[derive(Debug)]
-struct Padding(u64, u64, u64, u64);
+#[cfg(not(feature = "nightly"))]
+struct Inner<T> {
+    bytes: [u8; CACHE_LINE_BYTES],
+    _marker: PhantomData<T>,
+}
 
-/// Pad `T` to the length of a cacheline.
-///
-/// Sometimes concurrent programming requires a piece of data to be padded out
-/// to the size of a cacheline to avoid "false sharing": cachelines being
-/// invalidated due to unrelated concurrent activity. Use the `CachePadded` type
-/// when you want to *avoid* cache locality.
+#[cfg(feature = "nightly")]
+#[repr(align(64))]
+#[allow(unions_with_drop_fields)]
+union Inner<T> {
+    bytes: [u8; CACHE_LINE_BYTES],
```

Comment:
May I ask why do we need it here?

Reply:
Good point, looks like `align` is enough in nightly (https://play.rust-lang.org/?gist=4c3841431f1d27ff7997eb4f2c2a2471&version=nightly) so I think we can get away without the union at all.